src/main/scala/arithexpr/arithmetic/ArithExpr.scala

```diff
@@ -1818,6 +1817,8 @@ object Var {
 }
 
 class NamedVar (override val name: String, override val range: Range = RangeUnknown, override val fixedId: Option[Long] = None) extends Var(name, range, fixedId) {
+  def withRange(r : Range) : NamedVar = NamedVar(name, r)
```

Comment:
actually, should this not be a `Var` function?

Reply:
yes! moved it there